src/NATS.Client.JetStream/NatsJSConsumer.cs

```diff
@@ -256,13 +256,18 @@ await foreach (var jsMsg in fc.Msgs.ReadAllAsync(cancellationToken))
         await _context.Connection.SubAsync(sub: sub, cancellationToken);
 
         await sub.CallMsgNextAsync(
-            new ConsumerGetnextRequest
-            {
-                Batch = max.MaxMsgs,
-                MaxBytes = max.MaxBytes,
-                IdleHeartbeat = timeouts.IdleHeartbeat.ToNanos(),
-                Expires = timeouts.Expires.ToNanos(),
-            },
+            opts.NoWait
+
+                // When no wait is set we don't need to send the idle heartbeat and expiration, because if no message is available nats doesn't send 404 instantly
```

Comment:
should this say `because if no message is available nats sends 404 instantly` ?  It says `doesn't send`

Reply:
oh yeah, true! I change this